build.gradle

```diff
@@ -17,16 +17,16 @@ test {
 }
 
 dependencies {
-    implementation 'com.google.code.gson:gson:2.8.9'
+    implementation 'com.google.code.gson:gson:2.9.0'
```

Comment:
Hey @BartoszLitwiniuk, thank you for tackling this.

Should you maybe consider a lighter serialization library? Or maybe manual serialization (no use of 3rd-party libraries)? It's always better to keep the backtrace-java library as slim as possible as importing other libraries in it will increase dependencies on the top projects.

Reply:
Updating gson will not solve the issue: https://github.com/google/gson/issues/1898

Reply:
Hey @mk-ahmed-saucelabs , thanks for you feedback! 

I understand your approach and your concerns about importing GSON and mostly agree with them.

In the past when adding this dependency, I was mainly guided by the serialization/deserialization performance, and according to the benchmarks, GSON is the fastest when serializing both small and large files. BTW correct me if I'm wrong but GSON doesn't any extra dependencies which should heavily impact our library. I don't think that writing custom own serialization library is a good idea because it sounds a bit like reinventing the wheel, it may has a problems with performance, handling edge cases and it will be harder to maintain custom serializer/deserializer.

I think that we can merge this pull-request because it is worth to upgrade gradle and dependencies despite the error you found. Then we can discuss and plan with @vlussenburg how to resolve your issue or/and replace json serialization library.